golem-rib/src/parser/literal.rs

```diff
@@ -169,36 +140,46 @@ mod tests {
     #[test]
     fn test_literal_with_interpolation11() {
         let input = "\"foo-${bar}-baz\"";
-        let result = rib_program()
-            .easy_parse(position::Stream::new(input))
-            .map(|x| x.0);
+        let result = Expr::from_text(input).unwrap();
         assert_eq!(
             result,
-            Ok(Expr::concat(vec![
+            Expr::concat(vec![
                 Expr::literal("foo-"),
                 Expr::identifier("bar"),
                 Expr::literal("-baz"),
-            ]))
+            ])
         );
     }
 
     #[test]
     fn test_interpolated_strings_in_if_condition() {
         let input = "if foo == \"bar-${worker_id}\" then 1 else \"baz\"";
-        let result = rib_expr().easy_parse(input);
+        let result = Expr::from_text(input).unwrap();
         assert_eq!(
             result,
-            Ok((
-                Expr::cond(
-                    Expr::equal_to(
-                        Expr::identifier("foo"),
-                        Expr::concat(vec![Expr::literal("bar-"), Expr::identifier("worker_id")])
-                    ),
-                    Expr::untyped_number(1f64),
-                    Expr::literal("baz"),
+            Expr::cond(
+                Expr::equal_to(
+                    Expr::identifier("foo"),
+                    Expr::concat(vec![Expr::literal("bar-"), Expr::identifier("worker_id")])
                 ),
-                ""
-            ))
+                Expr::untyped_number(1f64),
+                Expr::literal("baz"),
+            )
+        );
+    }
+
+    #[test]
+    fn test_interpolated_strings_with_special_chars() {
+        let input = "\"\n\t<>/!@#%&^&*()_+[]; ',.${bar}-ba!z-${qux}\"";
```

Comment:
It could have been property based test but there is already a GitHub issue to tackle this properly - that will give us bandwidth to decide how to tackle this properly.